src/sos/workflow_executor.py

```diff
@@ -1646,7 +1645,7 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
                             f'{len(sections)} pending step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
                         ),
                     )
-                    raise exec_error
+                    raise e
```

Comment:
Please revert this one to `raise exec_error` or change to `raise exec_error from e` since `exec_error` is what is supposed to be returned.